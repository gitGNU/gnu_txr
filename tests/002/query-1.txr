@#
@# This file is in the public domain.
@# It was authored by Kaz Kylheku <kkylheku@gmail.com> in 2009
@#
@(next)!ls @TESTDIR/proc | sort -n
@(collect)
@{process /[0-9]+/}
@  (next)@TESTDIR/proc/@process/status
Name:	@name
State:	@state (@state_desc)
SleepAVG:	@sleep_avg%
Tgid:	@tgid
Pid:	@proc_id
PPid:	@parent_id
@(bind pid proc_id)
@(bind ppid parent_id)
@(skip)
Uid:	@uid	@/.*/
Gid:	@gid	@/.*/
@  (next)$@TESTDIR/proc/@process/task
@  (collect)
@thr
@  (end)
@  (bind thread thr)
@  (some)
@    (next)@TESTDIR/etc/passwd
@    (skip)
@user:@pw:@uid:@/.*/
@  (and)
@    (bind user uid)
@  (end)
@(end)
@(output)
USER       PID  PPID S NAME             THREADS
@  (repeat)
@{user 8} @{proc_id -5} @{parent_id -5} @state @{name 16} @(rep)@thr, @(first)@(last)@thr@(single)-@(end)
@  (end)
@(end)
