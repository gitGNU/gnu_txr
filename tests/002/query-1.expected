USER       PID  PPID S NAME             THREADS
root         1     0 S init             -
root         2     1 S ksoftirqd/0      -
root         3     1 S events/0         -
root         4     3 S khelper          -
root         5     3 S kacpid           -
root        16     3 S kblockd/0        -
root        17     1 S khubd            -
root        28     1 S kswapd0          -
root        29     3 S aio/0            -
root       103     1 S kseriod          -
root       175     1 S scsi_eh_0        -
root       186     1 S kjournald        -
root       870     1 S udevd            -
root      1068     1 S dhclient         -
root      1235     1 S kjournald        -
root      1236     1 S kjournald        -
root      1620     1 S syslogd          -
root      1624     1 S klogd            -
rpc       1645     1 S portmap          -
rpcuser   1665     1 S rpc.statd        -
root      1698     1 S rpc.idmapd       -
root      1766     1 S vmware-guestd    -
root      1790     1 S rpciod           -
root      1791     1 S lockd            -
root      1821     1 S ypbind           1826, 1821
root      1839     1 S acpid            -
root      1851     1 S cupsd            -
root      1887     1 S sshd             -
root      1902     1 S xinetd           -
root      1921     1 S rpc.rquotad      -
root      1925     1 S nfsd             -
root      1926     1 S nfsd             -
root      1927     1 S nfsd             -
root      1928     1 S nfsd             -
root      1929     1 S nfsd             -
root      1930     1 S nfsd             -
root      1931     1 S nfsd             -
root      1932     1 S nfsd             -
root      1936     1 S rpc.mountd       -
root      1963     1 S crond            -
xfs       1989     1 S xfs              -
root      2008     1 S atd              -
dbus      2027     1 S dbus-daemon-1    2027
root      2041     1 S cups-config-dae  -
root      2052     1 S hald             -
root      2062     1 S mingetty         -
root      2124     1 S gdm-binary       -
root      2184  2124 S gdm-binary       -
root      2354  2184 S X                -
524       2551  2184 S gnome-session    -
524       2579     1 S ssh-agent        -
524       2625     1 S dbus-launch      -
524       2626     1 S dbus-daemon-1    2627
524       2631     1 S gconfd-2         -
524       2634     1 S gnome-keyring-d  -
524       2636     1 S bonobo-activati  -
524       2638     1 S gnome-settings-  -
524       2644     1 S gam_server       -
524       2661     1 S xscreensaver     -
524       2685     1 S metacity         -
524       2689     1 S gnome-panel      -
524       2691     1 S nautilus         2715, 2691, 2708, 2711, 2714, 2713, 2709, 2710, 2696
524       2693     1 S gnome-volume-ma  -
524       2695     1 S eggcups          -
524       2698     1 S gnome-vfs-daemo  2698
524       2701     1 S pam-panel-icon   -
524       2707     1 R rhn-applet-gui   -
524       2717  2701 S pam_timestamp_c  -
524       2718     1 S mapping-daemon   -
524       2720     1 S wnck-applet      -
524       2722     1 S mixer_applet2    -
524       2726     1 S clock-applet     -
524       2728     1 S notification-ar  -
524      15812 32675 S ssh              -
524      16248 31908 S su               -
root     16249 16248 S bash             -
root     27121     3 S pdflush          -
root     27243     3 S pdflush          -
root     27682  1887 S sshd             -
524      27684 27682 S sshd             -
524      27685 27684 S bash             -
root     29840 16249 R bash             -
root     30737     1 S dhclient         -
root     31905  1887 S sshd             -
524      31907 31905 S sshd             -
524      31908 31907 S bash             -
root     32672  1887 S sshd             -
524      32674 32672 S sshd             -
524      32675 32674 S bash             -
